components/tests/ui/testcases/web/create_scenario.txt

```diff
@@ -0,0 +1,298 @@
+    [Arguments]                             ${RootId}                           ${ProjectName}
+    Right Click Create P/D/S                ${RootId}       Project             ${ProjectName}
+    ${projectId}=                           Check Right And Center Panels For Active Container                      Project             ${ProjectName}
+    Project Should Exist                    ${projectId}
+
+Project Should Exist
+    [Arguments]                             ${projectId}    
+    ${nodeId}=                              Wait for Project Node               ${projectId}
+    Wait Until Page Contains Element        xpath=//li[@id='${treeRootId}']/ul/li[@id='${nodeId}']
+
+Create Dataset Using Icon and Check If Orphaned
+    [Arguments]                             ${isOrphaned}
+    ${projectId}=                           Run Keyword If                      '${isOrphaned}' == '${notOrphaned}' Get Text            css=tr.data_heading_id strong
+    ${datasetId}=                           Create Dataset                      ${CtxDatasetName}
+    Run Keyword If                          '${isOrphaned}' == '${orphaned}'    Dataset Exists and Is Orphaned      ${datasetId}
+    Run Keyword If                          '${isOrphaned}' == '${notOrphaned}' Dataset Exists and Is Not Orphaned  ${datasetId}        ${projectId}
+
+Create Dataset Using Right Click and Check If Orphaned
+    [Arguments]                             ${RootId}       ${DatasetName}      ${isOrphaned}
+    ${projectId}=                           Run Keyword If                      '${isOrphaned}' == '${notOrphaned}' Get Text            css=tr.data_heading_id strong
+    Right Click Create P/D/S                ${RootId}       Dataset             ${DatasetName}
+    ${datasetId}=                           Check Right And Center Panels For Active Container                      Dataset             ${DatasetName}
+    Run Keyword If                          '${isOrphaned}' == '${orphaned}'    Dataset Exists and Is Orphaned      ${datasetId}
+    Run Keyword If                          '${isOrphaned}' == '${notOrphaned}' Dataset Exists and Is Not Orphaned  ${datasetId}        ${projectId}    
+
+Dataset Exists and Is Orphaned
+    [Arguments]                             ${datasetId}    
+    ${nodeId}=                              Wait for Dataset Node               ${datasetId}
+    Wait Until Page Contains Element        xpath=//li[@id='${treeRootId}']/ul/li[@id='${nodeId}']
+
+Dataset Exists and Is Not Orphaned
+    [Arguments]                             ${datasetId}                        ${projectId}
+    ${datasetnodeId}=                       Wait for Dataset Node               ${datasetId}
+    ${projectnodeId}=                       Wait for Project Node               ${projectId}
+    Wait Until Page Contains Element        xpath=//li[@id='${treeRootId}']/ul/li[@id='${projectnodeId}']/ul/li[@id='${datasetnodeId}']
+
+Dataset Should Contain Image 
```

Comment:
This should probably take a ```DatasetId``` keyword to make it more generally useful (as you suggested). Then all you'd need to do it click the Dataset in the tree and wait until the Image appears as a child.
The previous keyword in the test below ```Create Dataset Using Right Click and Check If Orphaned``` could return the ```DatasetId``` so you wouldn't have to look it up again.